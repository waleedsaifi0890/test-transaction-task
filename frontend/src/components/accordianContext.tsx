import React, {
  useState,
  useContext,
  useCallback,
  useMemo,
  createContext,
} from 'react';
import { isNil, reject, equals } from 'remeda';
import { ChevronUpIcon, ChevronDownIcon } from '@heroicons/react/24/solid';

type ContainerContext = {
  toggleActivePanel: (label: string) => void;
  activePanels: string[];
};

type ExpandMode = 'single' | 'multiple';

type ContainerProps = {
  expandMode?: ExpandMode;
  children: JSX.Element[];
};

type PanelProps = {
  children: JSX.Element | JSX.Element[];
  label: string;
  icon?: React.ElementType;
};

const containerContext = createContext<ContainerContext>({
  toggleActivePanel: () => null,
  activePanels: [],
});

const Container = ({ children, expandMode = 'single' }: ContainerProps) => {
  const [activePanels, setActivePanels] = useState<string[]>([]);

  const toggleActivePanel = useCallback(
    (label: string) => {
      if (expandMode === 'single') {
        setActivePanels(activePanels.includes(label) ? [] : [label]);
      }

      if (expandMode === 'multiple') {
        setActivePanels(
          activePanels.includes(label)
            ? reject(activePanels, equals(label))
            : [...activePanels, label]
        );
      }
    },
    [activePanels, expandMode]
  );

  const context = useMemo(
    () => ({
      toggleActivePanel,
      activePanels,
    }),
    [toggleActivePanel, activePanels]
  );

  return (
    <section className="space-y-3">
      <containerContext.Provider value={context}>
        {children}
      </containerContext.Provider>
    </section>
  );
};

const generatePanelButtonClasses = (isActive: boolean) => `
	flex
	w-full
	items-center
	space-x-2
	flex-grow
	text-left
	subheading
	py-2
	border-b-2
	${isActive ? 'border-true-gray-400' : 'border-true-gray-200'}
	${isActive ? '' : 'text-true-gray-400'}
`;

const Panel = ({ children, label, icon: Icon }: PanelProps) => {
  const { toggleActivePanel, activePanels } = useContext(containerContext);

  const togglePanel = useCallback(() => {
    toggleActivePanel(label);
  }, [toggleActivePanel, label]);

  const isActive = useMemo(
    () => activePanels.includes(label),
    [activePanels, label]
  );

  const classes = useMemo(
    () => ({
      button: generatePanelButtonClasses(isActive),
    }),
    [isActive]
  );

  return (
    <article>
      <header>
        <button
          key={label}
          className={classes.button}
          type="button"
          onClick={togglePanel}
        >
          {!isNil(Icon) && <Icon className="w-5" />}
          <span>{label}</span>
          {isActive && <ChevronUpIcon className="!ml-auto w-5" />}
          {!isActive && <ChevronDownIcon className="!ml-auto w-5" />}
        </button>
      </header>
      <div className="mt-4">{isActive && children}</div>
    </article>
  );
};

const Accordion = {
  Container,
  Panel,
};

export default Accordion;

export type { ContainerContext, ExpandMode, ContainerProps, PanelProps };
