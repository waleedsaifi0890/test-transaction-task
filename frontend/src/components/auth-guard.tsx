import React, { FC, ReactNode, useEffect, useState } from 'react';
import { Navigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import api from '../utils/api'; 

interface AuthGuardProps {
  children: ReactNode;
}

const AuthGuard: FC<AuthGuardProps> = ({ children }): JSX.Element => {
  const [isAuthenticated, setIsAuthenticated] = useState<boolean | null>(null);
  const [isLoading, setIsLoading] = useState(true); // NEW
  const [isPlan,setIsPlan]=useState<boolean | null>(true);

  useEffect(() => {
    const initialize = async () => {
      setIsLoading(true); // NEW
      const token = window.localStorage.getItem('accessToken');
      if (!token) {
        setIsAuthenticated(false);
      } else {
        try {
          const response = await api.post('/auth/verify', { token });
          console.log('Response from /auth/verify:', response.data.plan);
          
          if (response.status === 200) {
            if(response.data.status===""){
              setIsPlan(false)
            }
            setIsAuthenticated(true);
          } else {
            window.localStorage.removeItem('accessToken');
            setIsAuthenticated(false);
          }
        } catch (err) {
          console.error(err);
          window.localStorage.removeItem('accessToken');
          setIsAuthenticated(false);
        }
      }
      setIsLoading(false); // NEW
    };
    initialize();
  }, []); // Empty array means only run once on mount

  console.log('Is authenticated log outside useEffect:', isAuthenticated);
  
 // If we are loading or isAuthenticated is null, do not render anything
 if (isLoading || isAuthenticated === null) {
  return <></>; // or your loading spinner here
}

// If we are not authenticated, navigate to root.
if (!isAuthenticated) {
  return <Navigate to="/" />;
}

// check plan subscription 
// if(!isPlan){
//   return <Navigate to="/pic-plan" />;
// }


// If we are authenticated, render children.
return <>{children}</>;
};

AuthGuard.propTypes = {
  children: PropTypes.node,
};

export default AuthGuard;
