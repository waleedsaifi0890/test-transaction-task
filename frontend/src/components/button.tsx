import React, { forwardRef, useMemo } from 'react';
import { isNil } from 'remeda';
import { collapse } from '@growthops/ext-ts';


type Variant = 'primary' | 'secondary' | 'tertiary';

type Size = 'small' | 'regular' | 'large';

type IconAlignment = 'left' | 'right';

type Icon = {
  svg: React.ElementType;
  alignment?: IconAlignment;
  className?: string;
};

type BaseButtonProps = {
  label: string;
  className?: string;
  variant?: Variant;
  size?: Size;
  icon?: Icon;
};

type SemanticButtonProps = BaseButtonProps &
  React.ComponentPropsWithoutRef<'button'>;

type LinkButtonProps = BaseButtonProps & React.ComponentPropsWithoutRef<'a'>;

const baseClasses = 'inline-flex items-center rounded !leading-none';

const variantClasses: Record<Variant, string> = {
  primary: 'bg-[#FFF200] text-black',
  secondary: 'bg-[#1C1E1F] text-white',
  tertiary: 'bg-[#A89898] text-white',
};

const sizeClasses: Record<Size, string> = {
  large: 'px-8 py-5 heading-six space-x-4',
  regular: 'px-3 py-4  text-[16px] font-semibold  space-x-3',
  small: 'px-5 py-3 text-small space-x-2',
};

const iconClasses: Record<Size, string> = {
  large: 'w-7',
  regular: 'w-5',
  small: 'w-4',
};

const generateIcon = (
  icon: Icon | undefined,
  alignment: IconAlignment,
  classes: string,
  alignmentAdjustmentClasses: string
) => {
  if (!isNil(icon) && (icon?.alignment ?? 'left') === alignment) {
    return (
      <icon.svg className={collapse([classes, alignmentAdjustmentClasses])} />
    );
  }
};

const useClasses = (
  variant: Variant,
  size: Size,
  className: string | undefined,
  icon: Icon | undefined
) =>
  useMemo(
    () => ({
      root: collapse([
        baseClasses,
        variantClasses[variant],
        sizeClasses[size],
        className ?? '',
      ]),
      icon: collapse([iconClasses[size], icon?.className ?? '']),
    }),
    [variant, size, className, icon?.className]
  );

const useIcons = (icon: Icon | undefined, classes: string) =>
  useMemo(
    () => ({
      left: generateIcon(icon, 'left', classes, '-ml-2'), // -ml-* here to visually weight the icon appropriately.
      right: generateIcon(icon, 'right', classes, '!-mr-2'), // !-mr-* here to visually weight the icon appropriately.
    }),
    [icon, classes]
  );

const Semantic = ({
  label,
  variant = 'primary',
  size = 'regular',
  className,
  icon,
  ...intrinsicButtonProps
}: SemanticButtonProps): JSX.Element => {
  const classes = useClasses(variant, size, className, icon);
  const icons = useIcons(icon, classes.icon);

  return (
    <button className={classes.root} {...intrinsicButtonProps} type="submit">
      {icons.left}
      <span>{label}</span>
      {icons.right}
    </button>
  );
};

const Link = forwardRef<HTMLAnchorElement, LinkButtonProps>(
  (
    {
      label,
      variant = 'primary',
      size = 'regular',
      className,
      icon,
      ...intrinsicAnchorProps
    }: LinkButtonProps,
    ref
  ): JSX.Element => {
    const classes = useClasses(variant, size, className, icon);
    const icons = useIcons(icon, classes.icon);

    return (
      <a
        ref={ref}
        className={classes.root}
        {...intrinsicAnchorProps}
        rel="noreferrer"
      >
        {icons.left}
        <span>{label}</span>
        {icons.right}
      </a>
    );
  }
);

const Button = {
  Semantic,
  Link,
};

export default Button;

export type {
  Variant,
  Size,
  IconAlignment,
  Icon,
  BaseButtonProps,
  SemanticButtonProps,
  LinkButtonProps,
};
