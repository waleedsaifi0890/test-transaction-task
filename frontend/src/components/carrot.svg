<svg width="8" height="5" viewBox="0 0 8 5" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M8 2.44141e-06L4 4.76L-1.2207e-06 2.44141e-06H8Z" fill="black"/>
</svg>
