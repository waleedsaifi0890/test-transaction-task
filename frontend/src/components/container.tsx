import { collapse } from '@growthops/ext-ts';
import React, { useMemo } from 'react';

type ContainerProps = {
  children: boolean | JSX.Element | Array<boolean | JSX.Element>;
  className?: string;
};

const Container = ({ className = '', children }: ContainerProps) => {
  const classes = useMemo(
    () => ({
      root: collapse(['container', className]),
    }),
    [className]
  );

  return <div className={classes.root}>{children}</div>;
};

export default Container;

export type { ContainerProps };
