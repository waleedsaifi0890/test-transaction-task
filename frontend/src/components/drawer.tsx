import { sideImage } from '@app/data';
import React, { useState } from 'react';
import { useLocation, useNavigate } from 'react-router-dom';

const Drawer = ({ toggleDrawer, isOpen }: any) => {
  const navigate = useNavigate();
  const location =useLocation()
  const [isMenuOpen, setMenuOpen] = useState(false);
  const [menuIndex, setMenuIndex] = useState();


  return (
   
    <div className="md:hidden block">
    <div className="flex h-full w-[67px]">
      <nav className="bg-[#1C1D21] w-[67px] fixed  justify-between flex flex-col h-full z-20">
      <div className="mb-10">
          <div className="py-[15px] relative border-b-[#2D2F39] border-b-2 w-[40px] mx-auto">
            <img
              src="/images/Logo png.png"
              alt=""
              className="rounded-full w-8 h-12  mx-auto"
            />
            <div onClick={toggleDrawer} className={`bg-[#1c1e1f] ${isOpen?"":"hidden"} rounded-lg w-[26px] h-[26px] absolute right-[-30px] flex justify-center items-center`}>
              <svg
                xmlns="http://www.w3.org/2000/svg"
                fill="none"
                viewBox="0 0 24 24"
                strokeWidth={1.5}
                stroke="currentColor"
                className="w-4 h-4 text-white"
              >
                <path
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  d="M18.75 19.5l-7.5-7.5 7.5-7.5m-6 15L5.25 12l7.5-7.5"
                />
              </svg>
            </div>
          </div>
          <div className="mt-14">
            <ul>
              {sideImage.map((item, index) => (
                <div
                  className="w-full relative"
                >
                  <li  onClick={() => navigate(item.route)} className={`${
                        location.pathname === item.route || item.subPage===location.pathname  ? 'bg-[#000000]' : ''
                      } h-[40px] w-[44px] cursor-pointer hover:bg-[#000000] rounded-lg flex items-center justify-center mx-auto mb-[10px] relative`}>
                    <img
                      src={location.pathname===item.route || item.subPage===location.pathname ? item.img1:item.img}
                      alt=""
                      className="fill-current  text-gray-300  h-[20px] w-[20px]"
                    />
                    <div
                      className="absolute z-40 top-0 text-black bg-white hover:bg-[#7C8198] hover:text-white rounded-lg"
                      style={{
                        transform: 'translate(90px,0px)',
                        boxShadow: 'rgba(0, 0, 0, 0.24) 0px 3px 8px',
                      }}
                    >
                      <li>
                        {isMenuOpen && menuIndex === index && (
                          <ul className="py-2 px-4 w-32  flex cursor-pointer">
                            <li className="flex-wrap">Menu Item 1</li>
                          </ul>
                        )}
                      </li>
                    </div>
                  </li>
                  {location.pathname === item.route || item.subPage===location.pathname ?  (
                      <img
                        className="absolute right-[-2px] top-4"
                        src="/images/Polygon 1.svg"
                        alt=""
                      />
                    ):""}
                </div>
              ))}
            </ul>
          </div>
        </div>
       
      </nav>
      <div className=" py-4 text-[#1C1D21] bg-gray-200 h-screen "></div>
    </div>
  </div>
  );
};

export default Drawer;
