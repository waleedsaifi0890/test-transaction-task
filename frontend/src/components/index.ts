export { default as AuthGuard } from './auth-guard';
export { default as ProgressBar } from './progress-bar';
export { default as Accordion } from './accordianContext';
export { default as Button } from './button';
export { default as Card } from './card';
export { default as Container } from './container';
export { default as Heading } from './heading';
export { default as Input } from './input';
export { default as Region } from './region';
export { default as Select } from './select';
export { default as Tab } from './tab';
export { default as Text } from './text';
export { default as Textarea } from './textarea';
export { default as Layout} from "./layout";
export {default as Sidebar} from "./sidebar";
export {default as Header} from "./header";
export {default as Pagination} from "./pagination";
export {default as FullPageTable} from "./tabel";

