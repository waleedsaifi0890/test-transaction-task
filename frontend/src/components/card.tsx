import React, { useMemo } from 'react';
import { isNil } from 'remeda';
import { collapse } from '@growthops/ext-ts';
import { Heading, Text } from '@app/components';
import { CheckIcon } from '@heroicons/react/24/solid';

type CardProps = {
  image: string;
  image1: string;
  heading: string;
  heading4:string;
  content: string;
  contentGeneral:string;
  price: string;
  limit: string;
  information: any;
  isFramed?: boolean;
  isGeneral?: boolean;
  hasAutoHeight?: boolean;
  meta?: JSX.Element;
};

const generateRootClasses = (isFramed: boolean) => `
	overflow-hidden
	rounded
	h-full
	flex
	flex-col
	${isFramed ? 'bg-white shadow-lg' : ''}
  
`;

const generateBodyClasses = (isFramed: boolean) => `
	flex
	flex-col
	flex-grow
	${isFramed ? 'px-4 py-10' : ''}
`;

const generateMetaClasses = (hasAutoHeight: boolean) => `
	${hasAutoHeight ? 'mt-auto' : ''}
  mt-6
`;

const Card = ({
  image,
  image1,
  heading,
  heading4,
  content,
  contentGeneral,
  price,
  limit,
  meta,
  information,
  isGeneral,
  isFramed = false,
  hasAutoHeight = false,
}: CardProps) => {
  const classes = useMemo(
    () => ({
      root: collapse(generateRootClasses(isFramed)),
      image1: 'h-6 w-6',
      image: isFramed ? '' : 'rounded shadow-lg',

      body: collapse(generateBodyClasses(isFramed)),
      heading: '',
      heading4:"text-center",
      price: '',
      limit: 'text-[#6E7191] text-[16px] ml-2',
      content: 'mt-2 text-[#6E7191]',
      contentGeneral:"text-center !text-[13px] text-[#4E4B66] mt-2",
      meta: collapse(generateMetaClasses(hasAutoHeight)),
    }),
    [isFramed, hasAutoHeight]
  );

  return isGeneral ? (
   <div className='border border-[#D9DBE9] w-fit rounded-lg' style={{boxShadow: "0px 2px 22px rgba(0, 0, 0, 0.17), 0px 4px 30px rgba(7, 5, 5, 0.18)"}}>
      <img className={classes.image} src={image} />
      <div className={classes.body}>
      <div className='px-3 py-5'>
        <Heading
          className={classes.heading4}
          label={heading4}
          variant="heading-four"
        />
       
      
        <Text variant="text-small"  className={classes.contentGeneral} element="p" >
          {contentGeneral}
        </Text>
        </div>
      </div>
      </div>
  ) : (
    <article className={classes.root}>
      {/* <img className={classes.image} src={image} /> */}
      <div className={classes.body}>
        <Heading
          className={classes.heading}
          label={heading}
          variant="heading-four"
        />
        <Text className={classes.content} variant="text-regular">
          {content}
        </Text>
        <div className="flex items-center mt-4">
          <Heading label={price} variant="headline" element="p" />
          <sub className={classes.limit}>{limit}</sub>
        </div>
        {!isNil(meta) && <div className={classes.meta}>{meta}</div>}
        <div className="mt-8">
          {information.map((item: any) => (
            <div className="flex">
              <img className={classes.image1} src={image1} />
              <Text className="ml-2">{item.text}</Text>
            </div>
          ))}
        </div>
      </div>
    </article>
  );
};

export default Card;

export type { CardProps };
