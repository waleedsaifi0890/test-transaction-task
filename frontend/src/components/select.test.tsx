import React from 'react';
import { expect, test } from 'vitest';
import renderer from 'react-test-renderer';
import { Select } from '@app/components';

const selectOptions = [
  { key: '1', label: 'Daenerys Targaryen' },
  { key: '2', label: 'Jon Snow' },
  { key: '3', label: 'Arya Stark' },
  { key: '4', label: 'Brandon Stark' },
  { key: '5', label: 'Tyrion Lannister' },
];

test('snapshot — vanilla', () => {
  const result = renderer
    .create(<Select label="Character" options={selectOptions} />)
    .toJSON();

  expect(result).toMatchSnapshot();
});

test('snapshot — vanilla + required', () => {
  const result = renderer
    .create(
      <Select
        required
        label="Character"
        placeholder="Select a character..."
        options={selectOptions}
      />
    )
    .toJSON();

  expect(result).toMatchSnapshot();
});

test('snapshot — helptext', () => {
  const result = renderer
    .create(
      <Select
        required
        label="Character"
        placeholder="Select a character..."
        helpText="Who will become king or queen?"
        options={selectOptions}
      />
    )
    .toJSON();

  expect(result).toMatchSnapshot();
});

test('snapshot — disabled', () => {
  const result = renderer
    .create(
      <Select
        disabled
        label="Character"
        placeholder="Select a character..."
        options={selectOptions}
      />
    )
    .toJSON();

  expect(result).toMatchSnapshot();
});

test('snapshot — error', () => {
  const result = renderer
    .create(
      <Select
        required
        label="Character"
        placeholder="Select a character..."
        error="Please refer to season 8"
        options={selectOptions}
      />
    )
    .toJSON();

  expect(result).toMatchSnapshot();
});
