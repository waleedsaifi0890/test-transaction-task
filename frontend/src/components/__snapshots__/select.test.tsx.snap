// Vitest Snapshot v1

exports[`snapshot — disabled 1`] = `
<div>
  <span
    className="block text-small font-semibold "
  >
    Character
  </span>
  <div
    className="relative"
  >
    <svg
      aria-hidden="true"
      className="absolute top-0 right-0 mt-[0.9rem] mr-2 w-5 text-true-gray-500 select-none pointer-events-none"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fillRule="evenodd"
      />
    </svg>
    <select
      className="mt-1 border rounded px-2 py-1 w-full text-regular border-true-gray-300 bg-gray-200 cursor-not-allowed appearance-none"
      defaultValue=""
      disabled={true}
    >
      <option
        disabled={true}
        value=""
      >
        Select a character...
      </option>
      <option
        value="1"
      >
        Daenerys Targaryen
      </option>
      <option
        value="2"
      >
        Jon Snow
      </option>
      <option
        value="3"
      >
        Arya Stark
      </option>
      <option
        value="4"
      >
        Brandon Stark
      </option>
      <option
        value="5"
      >
        Tyrion Lannister
      </option>
    </select>
  </div>
</div>
`;

exports[`snapshot — error 1`] = `
<div>
  <span
    className="block text-small font-semibold text-red-500"
  >
    Character *
  </span>
  <div
    className="relative"
  >
    <svg
      aria-hidden="true"
      className="absolute top-0 right-0 mt-[0.9rem] mr-2 w-5 text-true-gray-500 select-none pointer-events-none"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fillRule="evenodd"
      />
    </svg>
    <select
      className="mt-1 border rounded px-2 py-1 w-full text-regular border-red-300 bg-white appearance-none"
      defaultValue=""
      required={true}
    >
      <option
        disabled={true}
        value=""
      >
        Select a character...
      </option>
      <option
        value="1"
      >
        Daenerys Targaryen
      </option>
      <option
        value="2"
      >
        Jon Snow
      </option>
      <option
        value="3"
      >
        Arya Stark
      </option>
      <option
        value="4"
      >
        Brandon Stark
      </option>
      <option
        value="5"
      >
        Tyrion Lannister
      </option>
    </select>
  </div>
  <div
    className="flex mt-1 items-center space-x-1 text-red-500"
  >
    <svg
      aria-hidden="true"
      className="w-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zM12 8.25a.75.75 0 01.75.75v3.75a.75.75 0 01-1.5 0V9a.75.75 0 01.75-.75zm0 8.25a.75.75 0 100-1.5.75.75 0 000 1.5z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="block text-xs"
    >
      Please refer to season 8
    </span>
  </div>
</div>
`;

exports[`snapshot — helptext 1`] = `
<div>
  <span
    className="block text-small font-semibold "
  >
    Character *
  </span>
  <div
    className="relative"
  >
    <svg
      aria-hidden="true"
      className="absolute top-0 right-0 mt-[0.9rem] mr-2 w-5 text-true-gray-500 select-none pointer-events-none"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fillRule="evenodd"
      />
    </svg>
    <select
      className="mt-1 border rounded px-2 py-1 w-full text-regular border-true-gray-300 bg-white appearance-none"
      defaultValue=""
      required={true}
    >
      <option
        disabled={true}
        value=""
      >
        Select a character...
      </option>
      <option
        value="1"
      >
        Daenerys Targaryen
      </option>
      <option
        value="2"
      >
        Jon Snow
      </option>
      <option
        value="3"
      >
        Arya Stark
      </option>
      <option
        value="4"
      >
        Brandon Stark
      </option>
      <option
        value="5"
      >
        Tyrion Lannister
      </option>
    </select>
  </div>
  <div
    className="flex mt-1 items-center space-x-1 text-true-gray-500"
  >
    <svg
      aria-hidden="true"
      className="w-4"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M2.25 12c0-5.385 4.365-9.75 9.75-9.75s9.75 4.365 9.75 9.75-4.365 9.75-9.75 9.75S2.25 17.385 2.25 12zm8.706-1.442c1.146-.573 2.437.463 2.126 1.706l-.709 2.836.042-.02a.75.75 0 01.67 1.34l-.04.022c-1.147.573-2.438-.463-2.127-1.706l.71-2.836-.042.02a.75.75 0 11-.671-1.34l.041-.022zM12 9a.75.75 0 100-1.5.75.75 0 000 1.5z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="block text-xs"
    >
      Who will become king or queen?
    </span>
  </div>
</div>
`;

exports[`snapshot — vanilla + required 1`] = `
<div>
  <span
    className="block text-small font-semibold "
  >
    Character *
  </span>
  <div
    className="relative"
  >
    <svg
      aria-hidden="true"
      className="absolute top-0 right-0 mt-[0.9rem] mr-2 w-5 text-true-gray-500 select-none pointer-events-none"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fillRule="evenodd"
      />
    </svg>
    <select
      className="mt-1 border rounded px-2 py-1 w-full text-regular border-true-gray-300 bg-white appearance-none"
      defaultValue=""
      required={true}
    >
      <option
        disabled={true}
        value=""
      >
        Select a character...
      </option>
      <option
        value="1"
      >
        Daenerys Targaryen
      </option>
      <option
        value="2"
      >
        Jon Snow
      </option>
      <option
        value="3"
      >
        Arya Stark
      </option>
      <option
        value="4"
      >
        Brandon Stark
      </option>
      <option
        value="5"
      >
        Tyrion Lannister
      </option>
    </select>
  </div>
</div>
`;

exports[`snapshot — vanilla 1`] = `
<div>
  <span
    className="block text-small font-semibold "
  >
    Character
  </span>
  <div
    className="relative"
  >
    <svg
      aria-hidden="true"
      className="absolute top-0 right-0 mt-[0.9rem] mr-2 w-5 text-true-gray-500 select-none pointer-events-none"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M12.53 16.28a.75.75 0 01-1.06 0l-7.5-7.5a.75.75 0 011.06-1.06L12 14.69l6.97-6.97a.75.75 0 111.06 1.06l-7.5 7.5z"
        fillRule="evenodd"
      />
    </svg>
    <select
      className="mt-1 border rounded px-2 py-1 w-full text-regular border-true-gray-300 bg-white appearance-none"
      defaultValue=""
    >
      <option
        value="1"
      >
        Daenerys Targaryen
      </option>
      <option
        value="2"
      >
        Jon Snow
      </option>
      <option
        value="3"
      >
        Arya Stark
      </option>
      <option
        value="4"
      >
        Brandon Stark
      </option>
      <option
        value="5"
      >
        Tyrion Lannister
      </option>
    </select>
  </div>
</div>
`;
